### ISSUE TITLE
Run Templater command

### ISSUE THREAD

**Issue by reply2za** [mProjectsCode/obsidian-meta-bind-plugin#461]
### Please fill out these Check-boxes

- [X] I checked for existing similar feature requests
- [X] I have read the docs and checked that the feature I am requesting is not already implemented
- [X] My feature request consists of only one feature

### Is your Feature Request related to a Problem or Annoyance?

I would like to run buttons as templater commands without creating new notes. As a workaround, I use `templaterCreateNote` and then I put a `return;` at the end of the templater file, but there are two issues with the method. One is that it often throws a silent error in the console, and the second issue is that it does briefly create a file in my file navigation and then only deletes it when the command is done executing. 

Another alternative is to give the templater file a templater hotkey, and then running that via the meta bind button's command action ([seen here](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/reference/buttonactions/command/)) but this is much more setup than having a dedicated feature.

### Describe the Feature you'd like

We should have a `runTemplaterFile` for this purpose to run the templater file without creating a note.

**Comment by mProjectsCode** [comment]
I am not familiar with templater commands. Can you point me to the relevant documentation or extensive examples?

**Comment by reply2za** [comment]
For example, in a templater file, I would have 

```js
const closeUnpinnedLeaves = () => {
	const leaves = [];
	app.workspace.iterateRootLeaves(leaf => { 
		leaves.push(leaf);
	});
	leaves.forEach(leaf => {
	    if (!leaf.pinned) {
	        leaf.detach();
	    }
	});
}

tp.user.closeUnpinnedLeaves();
return
```

And when I run this via templater's `Templater: Open insert template modal`, it won't create a new note but it runs the command. This makes templater really useful in quickly running some js commands with the addition of Templater's API. I added this to meta bind's `templaterCreateNote` to run such commands like closing all unpinned tabs, without actually creating a new note.

**Comment by reply2za** [comment]
Additional info 

```meta-bind-button
style: default
icon: trophy
label: "Close tabs
actions:
  - type: templaterCreateNote
    templateFile: "templates/commands/Close unpinned tabs.md"
    templater: true
```

**Comment by reply2za** [comment]
So I was thinking that this should be achievable within meta-bind. Mainly because when running a templater script in command format, through hotkeys or other non-meta-bind means, it doesn't create a new file. 

**Specifics on running a command outside of meta-bind without it showing a new file in the sidebar during execution**
- Create a templater file (include a return at the end of the script) 
- Add the file to the "Template Hotkeys" section within templater's settings
- Bind the command to a hotkey (doesn't matter if you use insert or create), within the hotkey settings
- End result is a command you can run (via the hotkey) without visibly creating a file   

As to your second point, unfortunately there's more setup to create a js file in and outside of Obsidian, so it's a bit harder to manage, and it doesn't include some of the useful note related functions provided by the `tp` namespace.

I hope you do consider it for these reasons.

**Comment by Graywaren** [comment]
I'm currently setting up meta bind buttons to run Templater template files and stumbled on to this thread. I wanted to chime in because, while I could be wrong, it seems like there's maybe some confusion on using templater alone and/or in combination with the buttons as they, rather than any changes needed in meta-bind. 

As long as your template is listed in the Templater Settings: Template Hotkeys list, you can choose it from the list of commands in a meta-bind button. When created from the right-click context menu you choose "run command" > add action > "change command" > then choosing the template from the list of commands. Your example template of closing unpinned tabs works and is easily creating as a mb button.

Also, you don't need to add "return;" at the end of your template, though it then matters whether you use "create" or "insert" if you're running it from a hotkey or a button. As long as you don't have blank lines after the closing tag or any YAML properties (unless your template is set up to ignore the properties) on your template, and use insert, the return is unnecessary. Here's examples using your close unpinned tabs template:
No return at the end:
<%* 
const closeUnpinnedLeaves = () => {
	const leaves = [];
	app.workspace.iterateRootLeaves(leaf => { 
		leaves.push(leaf);
	});
	leaves.forEach(leaf => {
	    if (!leaf.pinned) {
	        leaf.detach();
	    }
	});
}

closeUnpinnedLeaves();
-%>


Button using the "insert" option:
```meta-bind-button
label: This is a button
icon: ""
hidden: false
class: ""
tooltip: ""
id: ""
style: default
actions:
  - type: command
    command: templater-obsidian:[file path]/close tabs-without-return.md

```

This is the button with the script exactly as you wrote it where it doesn't matter if you use the insert or create version, my example uses create:
```meta-bind-button
label: This is a button
icon: ""
hidden: false
class: ""
tooltip: ""
id: ""
style: default
actions:
  - type: command
    command: templater-obsidian:create-[file path]/close tabs-with-return.md

```

Press either button and all non-pinned tabs are closed, no new file created, and nothing inserted into the current note. I did not see any new file show in the navigation tree during execution using either button, but the first option using insert without the return should definitely not be creating a new file during the execution, so I'd recommend using that.

Now, if that isn't what you're trying to do, maybe I'm misunderstanding what the request is? Is the request to run a templater command without having a separate .md file containing the template? As in, the entire template is contained within the code block of the button? If so, I can see that being useful for some short templater things, but the example mentioned of what the button would look like shows a link to a md file.

Is it a request to run a templater file without needing to add it to Templater's list of templates able to have a hotkey assigned? If so that would definitely be useful, if possible, because I dislike having templates I just want for a single button somewhere cluttering up that list and the hotkey list. I'd definitely love to see that feature.

Please forgive me if I've misunderstood something, but I've often spent hours trying to figure out a Templater thing only to realize I entirely misunderstood something (I won't tell you how long it took me to realize there were two options for the hotkeys: create & insert >.> ), so I wanted to save you some time if I could!

**Comment by reply2za** [comment]
Simply put, meta-bind's buttons are very useful in running templater commands without having to create a new note. So adding a `runTemplaterFile` button allows buttons to become powerful actions in a user's vault (shown in example I provided earlier). 

It's as if `templaterCreateNote` and `Run Javascript File` had a child resulting in benefits from both, and a plethora of use cases. 

@mProjectsCode 
There is already support to run/evaluate a templater file without creating a new note. This is likely used in the "replaceSelf" action as it does take a templater file as an optional argument. I believe that [this templater api function](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/blob/7440d58aaa4e55f5214aab493401d685ddf5c3c6/packages/obsidian/src/ObsidianInternalAPI.ts#L180) may be of use.

We can copy the [TemplaterCreateNoteButtonActionConfig](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/blob/7440d58aaa4e55f5214aab493401d685ddf5c3c6/packages/core/src/fields/button/actions/TemplaterCreateNoteButtonActionConfig.ts) class to create a new action. The only change being that we replace [createNoteWithTemplater](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/blob/7440d58aaa4e55f5214aab493401d685ddf5c3c6/packages/core/src/fields/button/actions/TemplaterCreateNoteButtonActionConfig.ts#L33) with [evaluateTemplaterTemplate](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/blob/7440d58aaa4e55f5214aab493401d685ddf5c3c6/packages/obsidian/src/ObsidianInternalAPI.ts#L180).

**Comment by mProjectsCode** [comment]
now that the PR has been merged this issue can be closed
